### ISSUE TITLE
Zeno doesn't start with get list and an empty line in list

### ISSUE THREAD

**Issue by CorentinB** [internetarchive/Zeno#146]
If you use get list with a seeds list that contain an empty line, Zeno won't start crawling.

**Comment by yzqzss** [comment]
https://github.com/internetarchive/Zeno/blob/cfa298060090e8d6c78662c3e01479df6713e720/internal/pkg/queue/item.go#L76


https://pkg.go.dev/net/url#Parse

> Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities. 

https://go.dev/play/p/gY82hrbRTh2

---

We probably need a better url parser.

**Comment by CorentinB** [comment]
> https://github.com/internetarchive/Zeno/blob/cfa298060090e8d6c78662c3e01479df6713e720/internal/pkg/queue/item.go#L76
> 
> https://pkg.go.dev/net/url#Parse
> 
> > Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities.
> 
> https://go.dev/play/p/gY82hrbRTh2
> 
> We probably need a better url parser.

Agreed. On the same topic.. I found Zeno to be absolutely unusable (I guess it's the queue) when queuing a 8.5M long seeds list. You might want to try it out if you're interested..

**Comment by akshithio** [comment]
Is `get list` a deprecated command of some sort? I'm not sure I see it on the `dev/v2` branch?

**Comment by yash-raj10** [comment]
Hey @CorentinB, is this issue still open?

I'm exploring the codebase and looking for issues to contribute to the repo and getting familiar with the code. Could you suggest something unassigned and open that would be a good starting point for me?

Thanks!

**Comment by CorentinB** [comment]
Hi, current development is on dev/v2, feel free to check if the issue still happen with it :)

**Comment by ParthAggarwal16** [comment]
Hi @CorentinB,
I’m exploring this issue to contribute. I noticed there’s no public dev/v2 branch in the repo — is that branch private, or has development already been merged back into main?
Just want to make sure I’m testing/fixing the issue on the correct branch.

**Comment by AltayAkkus** [comment]
The issue with empty/malformed seed URLs silently failing was apparently silently fixed :) The other issue also appears to be fixed, we could close this.
```
altay@Altays-MacBook-Pro Zeno % Zeno get url "" "https://www.france.fr"

2026/01/30 13:03:52 INFO User-Agent set to user-agent="Mozilla/5.0 (compatible; archive.org_bot +http://archive.org/details/archive.org_bot) Zeno/aa1fbdb warc/v0.8.96"
2026-01-30T13:03:52+01:00 INFO  reactor.go:57       | started component=reactor
...
2026-01-30T13:03:52+01:00 ERROR pipeline.go:153     | unable to parse seed component=controler.StartPipeline err=parse "": empty url
panic: parse "": empty url
```

https://github.com/internetarchive/Zeno/blob/aa1fbdb71fd2d5f0ee30509c08e2ec4d6bfe20a2/internal/pkg/controler/pipeline.go#L148-L154

`models.NewURL` calls `url.ParseRequestURI` which `err`'s out with `""` (and other malformed URLs, presumably)


All three seed URL ingestion paths (LQ, HQ and CLI) parse their respective URLs using `models.NewURL`
https://github.com/internetarchive/Zeno/blob/aa1fbdb71fd2d5f0ee30509c08e2ec4d6bfe20a2/internal/pkg/source/lq/consumer.go#L163-L166
https://github.com/internetarchive/Zeno/blob/aa1fbdb71fd2d5f0ee30509c08e2ec4d6bfe20a2/internal/pkg/source/hq/consumer.go#L165-L168

After parsing all seed URLs are send to the Reactor using `.receiveInsert()` and added to the `input` channel of the reactor, not a queue anymore. 

Since `input` holds all URLs which are to be crawled, it is safe to assume that @CorentinB  additional problem with 8.5M URLs clogging Zeno up is also fixed.
